docs/Developers/Design/CRD_design.md

```diff
@@ -97,154 +134,135 @@ are presented as a list in `replication.targets`, providing a `name` for each
+`maxNodes`, `freeStorageMin`, and `freeStorageMax` are configured. In this
+case, the number of stroage nodes always remains between the min and max, and
+scaling within that range is triggered based on the amount of free storage (not
+assigned to a brick) exists across the nodes in that template.
+
+Each template is likely to have a `nodeAffinity` entry to guide the placement
+of the Gluster pods to a single failure domain within the cluster.
+
+The `storage` block defines how the backing storage for the templated nodes are
+created. This includes the name of a StorageClass that can be used to allocate
+block-mode PVs, and the capacity that should be requested from this class.
 
 ## Node CR
 
-The Node CR's usage is slightly different than the Cluster and Zone objects.
-The Cluster and Zone objects are created by the administrator when configuring
-the cluster, but the Node objects are created and deleted dynamically by the
-operator.
+The Node CR defines a single Gluster server that is a part of the cluster.
+These node objects can either be created automatically by the operator from a
+template or they can be manually created.
 
 ```yaml
 apiVersion: "operator.gluster.org/v1alpha1"
-kind: Node
+kind: GlusterNode
 metadata:
   # Name for this node
   name: az1-001
-  # Nodes belong to a zone
-  zone: az1
-  # Nodes belong to a cluster also
-  cluster: my-cluster
   # CRD is namespaced
-  namespace: gluster
+  namespace: gcs
+  annotations:
+    # Applied by operator when it creates/manages this object from a template.
+    # When this is present, contents will be dynamically adjusted accd to the
+    # template in the cluster CR.
+    # When this annotation is present, the admin may only modify
+    # .spec.desiredState or delete the CR. Any other change will be
+    # overwritten.
+    anthill.gluster.org/template: template-name
 spec:
+  # Nodes belong to a cluster
+  cluster: my-cluster
+  # Nodes belong to a zone
+  zone: az1
   # Admin (or operator) sets desired state for the node.
-  targetState: active # (active | quiesce)
+  desiredState: enabled  # (enabled | disabled)
+  # Only 1 of external | storage
+  external:
+    address: my.host.com
+    credentials:
+      secretName: my-secret
+      secretNamespace: my-ns  # default is metadata.namespace
+  storage:
+    # Only 1 of device | pvcName
+    # Device names must to be stable on the host
+    - device: /dev/sd[b-d]
+      pvcName: my-pvc
+      tags: [tag1, tag2]
+  nodeAffinity:
+    # https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+    # For admin created GlusterNodes, this needs to specify a node selector
+    # that matches exactly one node. For template-based GNs, this will inherit
+    # from the template.
+    ...
 status:
   # TBD operator state
-  # Possible states: (active | deleting | quiesced)
-  currentState: active
+  # Possible states: (enabled | deleting | disabled)
```

Comment:
They are related, but not quite the same.

An admin may trigger the deletion of a gluster node by deleting the corresponding GlusterNode CR. The operator will have a finalizer on the object, preventing the actual removal until it is properly drained and removed from the cluster. The `currentState: deleting` is intended to signal that the operator has noticed the request to delete (or decided to do it on its own), and has started the process of draining/waiting/removing from cluster. The CR would actually be deleted (finalizer removed) only after the node and its resources are truly gone since the CR is the only way the operator can track the pending operations.